src/output.h

```diff
@@ -469,19 +469,48 @@ inline void popup_player_or_npc( player &p, const char *player_mes, const char *
 std::string format_item_info( const std::vector<iteminfo> &vItemDisplay,
                               const std::vector<iteminfo> &vItemCompare );
 
-input_event draw_item_info( const catacurses::window &win, const std::string &sItemName,
-                            const std::string &sTypeName,
-                            const std::vector<iteminfo> &vItemDisplay, const std::vector<iteminfo> &vItemCompare,
-                            int &selected, bool without_getch = false, bool without_border = false,
-                            bool handle_scrolling = false, bool scrollbar_left = true,
-                            bool use_full_win = false, unsigned int padding = 1 );
-
-input_event draw_item_info( int iLeft, int iWidth, int iTop, int iHeight,
-                            const std::string &sItemName, const std::string &sTypeName,
-                            const std::vector<iteminfo> &vItemDisplay, const std::vector<iteminfo> &vItemCompare,
-                            int &selected, bool without_getch = false, bool without_border = false,
-                            bool handle_scrolling = false, bool scrollbar_left = true,
-                            bool use_full_win = false, unsigned int padding = 1 );
+// the extra data that item_info needs to draw
+struct item_info_data {
+    private:
+        std::string sItemName;
+        std::string sTypeName;
+        std::vector<iteminfo> vItemDisplay;
+        std::vector<iteminfo> vItemCompare;
+    public:
+
+        item_info_data() = default;
+        item_info_data( const std::string &sItemName, const std::string &sTypeName,
+                        const std::vector<iteminfo> &vItemDisplay, const std::vector<iteminfo> &vItemCompare,
+                        const int selected = 0 )
+            : sItemName( sItemName ), sTypeName( sTypeName ),
+              vItemDisplay( vItemDisplay ), vItemCompare( vItemCompare ),
+              selected( selected ) {}
+
+        std::string get_item_name() const {
+            return sItemName;
+        }
+        std::string get_type_name() const {
+            return sTypeName;
+        }
+        std::vector<iteminfo> get_item_display() const {
+            return vItemDisplay;
+        }
+        std::vector<iteminfo> get_item_compare() const {
```

Comment:
Those getters could return const references. No need to copy the values every time.